multicluster/controllers/multicluster/commonarea/remote_common_area_test.go

```diff
@@ -79,9 +147,124 @@ func TestMemberAnnounce(t *testing.T) {
 
 	select {
 	case <-done:
-	case <-time.After(10 * time.Second):
+	case <-time.After(15 * time.Second):
 		panic("timeout")
 	}
 
 	remoteCommonAreaUnderTest.Stop()
+	status := remoteCommonAreaUnderTest.GetStatus()
+
+	assert.Equal(t, v1.ConditionTrue, status[0].Status)
+	assert.Equal(t, v1.ConditionTrue, status[1].Status)
+	assert.Equal(t, "", status[0].Reason)
+	assert.Equal(t, "", status[1].Reason)
+	assert.Equal(t, "", status[0].Message)
+	assert.Equal(t, "This leader cluster is the leader for local cluster", status[1].Message)
+	assert.Equal(t, mcsv1alpha1.ClusterConditionType(""), status[0].Type)
+	assert.Equal(t, mcsv1alpha1.ClusterReady, status[1].Type)
+}
+
+func TestMemberAnnounceNewRemoteCommonArea(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	mockManager := mocks.NewMockManager(mockCtrl)
+	fakeRemoteClient := fake.NewClientBuilder().WithScheme(scheme).Build()
+
+	expectedRemoteCommonArea := &remoteCommonArea{
+		Client:             fakeRemoteClient,
+		ClusterManager:     mockManager,
+		ClusterSetID:       "clusterSetA",
+		ClusterID:          "leaderA",
+		config:             nil,
+		scheme:             scheme,
+		Namespace:          "cluster-a-ns",
+		connected:          false,
+		localClusterClient: nil,
+		localNamespace:     "localnamespace",
+		localClusterID:     "clusterA",
+		clusterStatus: mcsv1alpha1.ClusterCondition{
+			Type:    mcsv1alpha1.ClusterReady,
+			Status:  v1.ConditionUnknown,
+			Message: "Leader cluster added",
+		},
+		leaderStatus: mcsv1alpha1.ClusterCondition{
+			Type:    mcsv1alpha1.ClusterIsLeader,
+			Status:  v1.ConditionFalse,
+			Message: "Leader cluster added",
+		},
+	}
+
+	actualRemoteCommonArea, err := NewRemoteCommonArea(expectedRemoteCommonArea.ClusterID, expectedRemoteCommonArea.ClusterSetID, expectedRemoteCommonArea.localClusterID, mockManager, fakeRemoteClient, scheme, nil,
+		"cluster-a-ns", "localnamespace", nil)
+	assert.Equal(t, nil, err)
+	clusterStatus, leaderStatus := actualRemoteCommonArea.GetStatus()[0], actualRemoteCommonArea.GetStatus()[1]
+	expectedRemoteCommonArea.clusterStatus.LastTransitionTime = clusterStatus.LastTransitionTime
+	expectedRemoteCommonArea.leaderStatus.LastTransitionTime = leaderStatus.LastTransitionTime
```

Comment:
These can be removed since you already compared them in following code.
`assert.Equal(t, expectedRemoteCommonArea, actualRemoteCommonArea)`

Reply:
this is nor for comparing, this is assignment, because NewRemoteCommonArea assign metav1.now() for these timestamps

Reply:
OK, it's better to add a comment about it's to simplify the comparison.